File: agent-driven-architecture-main/examples/typescript/src/blocks/artifact/adapter/adapter.ts
// ── blocks/artifact/adapter/adapter — THIS BLOCK'S IMPURE UNIT (G13) ──────
// The delivery client lives here and nowhere else. In a real deployment this
// writes to object storage or a ticketing system; here it writes lines to a
// caller-supplied emitter, so the demo runs offline.

import type { Emit } from "@adr/spine/pure/emit";
import type { DeliveryPort } from "../port";

export function liveDelivery(emit: Emit): DeliveryPort {
  return {
    deliver: (lines) => {
      emit(`[delivery] work product sealed, ${lines.length} line(s):`);
      lines.forEach((line) => emit(`  ${line.at} ${line.by}: ${line.text}`));
    },
  };
}

File: agent-driven-architecture-main/examples/kotlin/block/artifact/adapter/src/main/kotlin/adr/blocks/artifact/Adapter.kt
// ── blocks/artifact/adapter — the ONLY file in this block that may hold a client ─

package adr.blocks.artifact

import adr.contract.ArtifactResult.ArtifactLine
import adr.spine.pure.Emit

/**
 * The rim of the block. `write` stands in for the document store / mailer — a real
 * one would be a client handle held here and nowhere else.
 */
class LiveDelivery(private val write: Emit<String>) : DeliveryPort {
    override fun deliver(lines: List<ArtifactLine>) {
        write(lines.joinToString("\n") { "[${it.at.value}] ${it.by}: ${it.text}" })
    }
}
